### ISSUE TITLE
technical error

### ISSUE THREAD

**Issue by motoduder** [prairiesnpr/hass-tdameritrade#11]
Hello, I get the following "technical error" when trying to add the integration. Its from "https://auth.tdameritrade.com/auth?r*":

`This may be due to a technical error, or #{the client application} may be an attempt to fraudulently access your account. If you believe you’ve received this message in error, please contact the publisher of #{the client application}. For more information on securing your account, please see our security page.`

I dont think there is any kind of security breach but I am not sure what the issue could be. I will say that my experience when setting up the integration does not match the steps noted specifically around this part:

"You will be prompted to authenticate with TDAmeritrade, this requests an offline token and will be good for 90 days.
Once authenticated, select the TDAmeritrade integration and select settings.
Add your accounts as a comma seperated list and then click save."

For me this is just on one popup but its after this popup that I get the noted error. Any advice would be greatly appreciated. I am happy to provide any information that I am not providing not. Thank you! This definitely looks like an awesome project and I look forward to leveraging this for some project ideas I have had regarding finance.

**Comment by prairiesnpr** [comment]
Can you confirm the callback URL you configured for the API?  I think I saw this error when I had an unsecure URL set in testing.

**Comment by motoduder** [comment]
> Can you confirm the callback URL you configured for the API? I think I saw this error when I had an unsecure URL set in testing.

Absolutely the case. Overlooked a recent change on my end. Sorry about that. Thanks for you time!

**Comment by gilperme** [comment]
@prairiesnpr Thanks for this great add-on....unfortunately I am running into issues with setup.

I am using nginx proxy with valid ssl certs...I attempted the integration on two separate hassio installs and both received the same error. I am using externally accessible domains and have set up other integrations requiring a similar call back with no issues (Withings & Smarthings)

The URLS used are as follows:

instance 1: https://domain1.blah.com/auth/external/callback
instance 2: https://domain2.blah.com/auth/external/callback

I also verified the app URLs in the Dev portal before initiating each attempt.

In the proxy logs I see traffic being redirected to the correct hosts, but get the TDAmeritrade error each time. 

Any suggestions on nginx proxy settings that may help here? Or is there something I'm missing.

Thanks again!
